src/CatSearch.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react"
+import { useState, useEffect, useCallback } from "react"
+import { fetchSearchResultsFromAPI } from "./searchApiClient"
+
+export function CatSearch() {
+  const [valid, setValid] = useState(false)
+  const [hits, setHits] = useState<any[]>([])
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const searchTermOnChange = useCallback((event) => {
```

Comment:
I appreciate you thinking about optimizing performance here with `useCallback`. However, `useCallback` (like many hooks in React) can be a bit of a footgun, and this is one of those situations.

Because we have no dependency that would make caching this callback function worthwhile in-between re-renders, we will actually be allocating more memory in this component that we need to. Meaning that, with this use of `useCallback`, our component will actually be generally _slower_. 

`useCallback`, `useEffect` and `useMemo` are tricky to know when and when not to use. If you'd like to learn more, Kent C. Dodds has a great blogpost about this on his website with a ton of other React resources that I highly recommend https://kentcdodds.com/blog/usememo-and-usecallback.

If you have any questions or want more help understanding these concepts, please let me know. Happy to help.